dataservice/api/common/pagination.py

```diff
@@ -22,71 +23,90 @@ def paginated_wrapper(*args, **kwargs):
                 try:
                     after = parser.parse(after)
                 # Parser couldn't derive a datetime from the string
+                except ValueError:
+                    pass
+
+            # If the above was not able to parse as a date, try to parse a uuid
+            if not isinstance(after, datetime):
+                try:
+                    after = str(UUID(after))
                 except ValueError:
                     after = None
 
-        # Default to the unix epoch
+        # Begin pagination from the 0 uuid if there was no after param given
         if after is None:
-            after = datetime.fromtimestamp(0)
+            after = str(UUID(int=0))
 
-        return f(*args, **kwargs,
-                 after=after,
-                 limit=limit)
+        return f(*args, **kwargs, after=after, limit=limit)
 
     return paginated_wrapper
 
 
 class Pagination(object):
     """
-    Object to help paginate through endpoints using the created_at field
+    Object to help paginate through endpoints using the created_at field or
+    the uuid field
     """
 
     def __init__(self, query, after, limit):
-        assert type(after) is datetime
         self.query = query
         self.after = after
         self.limit = limit
         self.total = query.count()
         # Assumes that we only provide queries for one entity
         # This is safe as pagination only accesses one entity at a time
         model = query._entities[0].mapper.entity
-        assert hasattr(model, 'created_at')
-        self.items = (query.order_by(model.created_at.asc())
-                           .filter(model.created_at > after)
-                           .limit(limit).all())
+        # If the after param is datetime, order on the created_at column,
+        # otherwise, assume that after is a uuid and sort on the uuid column
+        order_by_field = (
+            model.created_at
+            if isinstance(after, datetime)
+            else model.uuid
+        )
+        self.items = (
+            query.order_by(order_by_field.asc())
+            .filter(order_by_field > after)
+            .limit(limit)
+            .all()
+        )
 
     @property
     def prev_num(self):
-        """ Returns the timestamp of the first item """
+        """ Returns the uuid of the first item """
```

Comment:
This function isn't used right now and could probably be removed until we support a `before` parameter.